===================================================================
RCS file: /cvs/cvsroot/smpeg/MPEGaudio.h,v
retrieving revision 1.24
retrieving revision 1.25
diff -u -r1.24 -r1.25
--- smpeg/MPEGaudio.h	2005/12/17 03:09:08	1.24
+++ smpeg/MPEGaudio.h	2005/12/17 03:17:06	1.25
@@ -155,12 +155,6 @@
 /* The actual MPEG audio class */
 class MPEGaudio : public MPEGerror, public MPEGaudioaction {
 
-    friend void Play_MPEGaudioSDL(void *udata, Uint8 *stream, int len);
-    friend int Play_MPEGaudio(MPEGaudio *audio, Uint8 *stream, int len);
-#ifdef THREADED_AUDIO
-    friend int Decode_MPEGaudio(void *udata);
-#endif
-
 public:
     MPEGaudio(MPEGstream *stream, bool initSDL = true);
     virtual ~MPEGaudio();
@@ -371,6 +365,20 @@
 #define N_TIMESTAMPS 5
 
   double timestamp[N_TIMESTAMPS];
+
+  /* Functions which access MPEGaudio internals */
+  friend void Play_MPEGaudioSDL(void *udata, Uint8 *stream, int len);
+  friend int Play_MPEGaudio(MPEGaudio *audio, Uint8 *stream, int len);
+#ifdef THREADED_AUDIO
+  friend int Decode_MPEGaudio(void *udata);
+#endif
 };
 
+/* Need to duplicate the prototypes, this is not a typo :) */
+void Play_MPEGaudioSDL(void *udata, Uint8 *stream, int len);
+int Play_MPEGaudio(MPEGaudio *audio, Uint8 *stream, int len);
+#ifdef THREADED_AUDIO
+int Decode_MPEGaudio(void *udata);
+#endif
+
 #endif /* _MPEGAUDIO_H_ */
